io.github.wivrn.wivrn.yml

```diff
@@ -0,0 +1,209 @@
+      tag: 3.4.0
+    cleanup:
+      - '*'
+
+  - name: nlohmann_json
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DJSON_BuildTests=OFF
+    sources:
+    - type: git
+      url: https://github.com/nlohmann/json.git
+      tag: v3.11.3
+
+  - name: cli11
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DBUILD_TESTING=OFF
+      - -DCLI11_BUILD_TESTS=OFF
+      - -DCLI11_BUILD_EXAMPLES=OFF
+    sources:
+    - type: git
+      url: https://github.com/CLIUtils/CLI11
+      tag: v2.4.2
+    cleanup:
+      - '*'
+
+  - name: wivrn
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DWIVRN_BUILD_CLIENT=OFF
+      - -DWIVRN_USE_VAAPI=ON
+      - -DWIVRN_USE_X264=ON
+      - -DWIVRN_USE_NVENC=ON
+      - -DWIVRN_USE_PULSEAUDIO=OFF
+      - -DWIVRN_USE_GLIB=ON
+      - -DFETCHCONTENT_FULLY_DISCONNECTED=ON
+      - -DFETCHCONTENT_BASE_DIR=deps
+      - -DXRT_IPC_MSG_SOCK_FILENAME=wivrn/comp_ipc
+      - -Wno-dev
+    sources:
+      - type: git
+        url: https://github.com/WiVRn/WiVRn.git
+        tag: 6da658bcfb4f67abd6c697f2c222529a20ac10ca # v0.18 + patches for flatpak not yet merged in master
+      - type: archive
+        url: https://github.com/boostorg/pfr/archive/refs/tags/2.2.0.tar.gz
+        dest: deps/boostpfr-src
+        sha256: 6f490ec1383de5855dcf801c23f0a967f888eb6e1440b6ca0bf36c12d4ae7ac7
+      - type: git
+        url: https://gitlab.freedesktop.org/monado/monado
+        tag: dfc602288ab05131584a3f2be18031a13fccd061
+        dest: deps/monado-src
+      - type: shell
+        commands:
+         - cd deps/monado-src && ../../patches/apply.sh ../../patches/monado
+    build-commands:
+      - install -DTm644 images/wivrn.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.wivrn.wivrn.svg
+      - install -Dm644 flatpak/io.github.wivrn.wivrn.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
+      - install -Dm644 flatpak/io.github.wivrn.wivrn.desktop -t ${FLATPAK_DEST}/share/applications
+
+  - name: opencomposite
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DUSE_SYSTEM_OPENXR=OFF
+      - -DUSE_SYSTEM_GLM=OFF
+    sources:
+      - type: git
+        url: https://gitlab.com/znixian/OpenOVR.git
+        tag: 632e5cc50b913e93194ca2970e6f13021182579f
+      - type: patch
+        paths:
+          - opencomposite-0001-Add-install-target.patch
+
+  - name: launcher
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 launch-wivrn-server.sh ${FLATPAK_DEST}/bin/launch-wivrn-server
+    sources:
+      - type: script
+        dest-filename: launch-wivrn-server.sh
+        commands:
+        - rm -f $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
+        - sed -ne 's#app-path=\(.*\)#{"runtime":["\1/OpenComposite"],"version":1}#p' /.flatpak-info > $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
+        - chmod -w $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
```

Comment:
We have to consider 2 issues:

When Steam starts, it will overwrite the content of `$XDG_CONFIG_HOME/openvr/openvrpaths.vrpath` to set itself as the OpenVR runtime. Marking the file as read-only prevents Steam from overwriting the file.

In case of flatpak update, the path will no longer be valid, which is the reason for `rm -f` of the old one.

`sed` is just being used to generate the file from `/.flatpak-info`, just a fancy copy.

We can remove the file after `wivrn-server` returns (there is no UI yet so user has to send Ctrl+C to exit), but I can't see a much better solution than `rm -f; sed` for setting the OpenVR runtime.